packages/react-strict-dom/tests/typed-om/CSSParser-test.js

```diff
@@ -0,0 +1,83 @@
+/**
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import CSSParser, { CSSFunction, CSSPreservedToken } from '../src/CSSParser';
+
+describe('CSSParser', () => {
```

Comment:
Help me understand the benefits of a custom parser over using the postcss parsers that we used in stylex?

Reply:
The `postcss-value-parser` is more like a tokenizer, but it should work well for parsing variables.

I also have a work-in-progress parser here: https://github.com/nmn/css-value-parser

Also, LightningCSS is a very good parser.

Reply:
This started as an exploration to better understand what the tokenizer/parser were actually doing — with a focus on targeted correctness over other parts, but I have additional ideas for improvements that if we went down the path of using postcss's value parser could keep me from making those improvements as the postcss value parser is unmaintained.

Ultimately I don't care enough to keep pushing tho so if you feel strongly enough I can rewrite this to use the postcss value parser.

Reply:
Not trying to push for one or the other, trying to gather information to understand the cost/benefit of adding code we maintain vs using code we don't maintain. What are some of the additional improvements you have in mind?

Reply:
@vincentriemer Other than LightningCSS I found all other parsers for CSS values lacking, including whatever parts of the CSS Typed OM exist so far so I understand the need for a custom parser. 

In order to avoid duplication of work in the future, let's figure out a way to combine your work and the parser I've been working on.

Reply:
@necolas The sort of thing I was thinking in terms of improvements was — because this sort of thing is so low level and we have strict flow safety — I can refactor the work to utilize tuples instead of objects for the token representation which is something I did in instyle back in the day to improve performance. Not to mention us owning our own parser lets us prioritize adding new parsing/tokenization features as we want them.

@nmn Yeah I agree w/ the avoiding of duplicating work but I've already migrated to use postcss-value-parser (pushing the changes soon) and it actually didn't require much refactoring so moving to a different parser (at least in the short term) won't be very difficult.